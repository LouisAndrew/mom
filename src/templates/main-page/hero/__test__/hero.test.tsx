import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Hero from '..';

describe('Hero element', () => {
    afterEach(cleanup);
    const el: React.ReactElement = <Hero />;
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
