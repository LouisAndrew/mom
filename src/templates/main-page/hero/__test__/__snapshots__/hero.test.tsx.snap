// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero element matches snapshot 1`] = `
<div
  className="sc-fzqBZW fMIXjC top"
>
  <div
    className="sc-fzqNJr cqsWMP"
    height={
      Array [
        "100%",
      ]
    }
    width={1}
  >
    <div
      id="hero-img"
    >
      <img
        alt="img"
        src="https://res.cloudinary.com/dsvdffre0/image/upload/v1596883624/sonnie-hiles-L0BaowhFe4c-unsplash_jwo8ie.jpg"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
    <div
      id="bg-shade"
    />
  </div>
  <div
    className="sc-AxgMl gWoUtq"
    display="flex"
  >
    <div
      className="sc-fznKkj dBOEcr"
    >
      <h1
        className="sc-AxjAm dQWbKG"
        color="dark.0"
        fontFamily="heading"
        fontSize={
          Array [
            5,
            5,
            6,
            6,
          ]
        }
        fontWeight="extraBold"
      >
        Siapa bilang cari rumah jaman sekarang susah?
      </h1>
      <p
        className="sc-AxhUy jFOsJE"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        width={1}
      >
        Cari rumah idamanmu sekarang! Cukup cantumkan kriteria mu, tanpa ribet
      </p>
      <form
        className="sc-fznZeY cniRBJ"
        color="dark.0"
        width={
          Array [
            1,
            1,
            0.6,
            0.5,
          ]
        }
      >
        <span
          className="select-span"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <label
            className="sc-fzokOt emYOXI"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            for="hero-form-location"
            htmlFor="hero-form-location"
            width={1}
          >
            Daerah
          </label>
          <div
            className="sc-AxheI eosWqM"
            width={1}
          >
            <label
              className="sc-AxmLO fPFTiL"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              for="hero-form-location"
              htmlFor="hero-form-location"
              width={
                Array [
                  1,
                ]
              }
            >
              
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
              <input
                id="hero-form-location"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </div>
        </span>
        <label
          className="sc-fzokOt fTZazG"
          display="flex"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          for="hero-form-area"
          htmlFor="hero-form-area"
          width={
            Array [
              "fit-content",
            ]
          }
        >
          Luas
          <div
            className="sc-fzplWN czaqLz"
          >
            <input
              className="sc-fzoLsD kDWymi"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              id="hero-form-area"
              onChange={[Function]}
              placeholder="100"
              type="number"
              width={
                Array [
                  52,
                  52,
                  64,
                ]
              }
            />
            <span
              className="sc-fzpans ewQpsq"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              id="hero-form-area"
              width={
                Array [
                  52,
                  52,
                  64,
                ]
              }
            >
              mÂ²
            </span>
          </div>
        </label>
        <span
          className="select-span type"
        >
          <label
            className="sc-fzokOt ilPzEh"
            display="flex"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            for="hero-form-type"
            htmlFor="hero-form-type"
            width={
              Array [
                1,
              ]
            }
          >
            Jual/Beli
          </label>
          <div
            className="sc-AxheI eosWqM"
            width={1}
          >
            <label
              className="sc-AxmLO emALWw"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              for="hero-form-type"
              htmlFor="hero-form-type"
            >
              
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
              <input
                id="hero-form-type"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </div>
        </span>
        <button
          className="sc-fznyAO hdmVWB"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          onClick={[Function]}
          width={
            Array [
              1,
            ]
          }
        >
          Cari Sekarang!
        </button>
      </form>
    </div>
  </div>
</div>
`;
